### PR TITLE
Add proxy mode

### PR DESCRIPTION
**Pull request by samouss**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -74,6 +74,15 @@ var defaults = {
...
     maps_dir: '../maps/'
+  },
+
+  // proxy mode config - !!! DON'T FORGET TO UPDATE BUILD DIR PATH AND PUT twig_dir TO NULL
+  proxy: {
+    // proxy url
+    url: 'http://127.0.0.1:8000',
+
+    // where to look for source files (relative to root)
+    source_dir: '../app/Resources/views/'
```
Comment (pitpit) [inline gulpfile.js]: replace `source_dir` by `watch_dir`

File: gulpfile.js [hunk 2]
```diff
@@ -217,6 +241,10 @@ gulp.task('start', ['browser-sync'], function() {
     gulp.watch(path.join(config.twig_dir, '**/*.{twig,json}'), ['twig']);
   }
 
+  if (argv['proxy'] !== undefined) {
+    gulp.watch(path.join(config.proxy.source_dir, '**/*.twig'), ['browser-sync-refresh']);
```
Comment (pitpit) [inline gulpfile.js]: replace `source_dir` by `watch_dir`

**Comment by pitpit** [comment]
When proxy.url is set, twig_dir and html_dir should be automatically set to null

File: gulpfile.js (hunk 1 above)
Comment (pitpit) [inline gulpfile.js]: and use a more restrictive pattern to watch only twig files

File: gulpfile.js (hunk 2 above)
Comment (pitpit) [inline gulpfile.js]: and remove specific twig pattern

Comment (samouss) [inline gulpfile.js]: If we remove specific twig pattern in this case, why do that on twig and html ?

**Comment by pitpit** [comment]
merci :+1:

**Pull request merged**